### PR TITLE
Spp 979 new logger

### PR DESCRIPTION
**Pull request by rhodriguerrier**
Have repointed most of the spp-engine to use the new spp logger instead of the logging.py class within the repo. It has not been added to write_queue_lambda.py and messaging.py though as these are not called anywhere in the engine (at least I couldn't find any calls) so I have no way of ensuring the parameters to create the logger are present.

### CODE REVIEW

File: spp/engine/pipeline.py [hunk 1]
```diff
@@ -215,39 +241,49 @@ def send_status(self, status, module_name, current_step_num=None):
                                       current_step_num=current_step_num,
                                       total_steps=len(self.methods))
 
-    def run(self, platform, crawler_name):
+    def run(self, platform, crawler_name,
+            survey, environment, run_id=run_id):
         """
         Runs the methods of the pipeline
```
Comment (glanvl) [inline spp/engine/pipeline.py:247]: Docstring

File: spp/engine/read.py [hunk 2]
```diff
@@ -42,63 +41,85 @@ def __repr__(self):
...
+def pandas_read(cursor, environment, run_id,
+                survey, reader=PandasReader(), **kwargs):
     """
     Reads data into a DataFrame using Pandas. If the cursor is an SPP Query,
     a database is queried and a
     PandasReader instance must be supplied which implements read_db(),
     otherwise the cursor string is treated like a file path.
     :param cursor: Query instance or file path String
     :param reader: DB connection object that extends PandasReader
+    :param environment: AWS environment name for spp logger
```
Comment (glanvl) [inline spp/engine/read.py:53]: Makes me think this is the loggers environment might be clearer to say - current running environment to be passed to the logger. Similarly with run_id and survey name.

File: spp/engine/read.py [hunk 3]
```diff
@@ -42,63 +41,85 @@ def __repr__(self):
...
+               run_id, survey, **kwargs):
     """
     Reads data into a DataFrame using Spark. If the cursor is an SPP Query,
     the Spark metastore is used,
     otherwise the cursor is treated like a file path.
     :param spark: Spark session
     :param cursor: Query instance or file path String
+    :param environment: AWS environment name for spp logger
+    :param run_id: Run_id string for spp logger
+    :param survey: Survey name for spp logger
```
Comment (glanvl) [inline spp/engine/read.py:84]: See comment above

**Review by glanvl — DISMISSED** [review]
Couple of minor changes mostly just with updating the docstring.

**Review by Piwington — APPROVED** [review]
Looks happy to me.

**Review by arthompson — APPROVED** [review]

**Pull request merged**